### PR TITLE
doc/glossary.rst: alphabetize glossary terms

### PR DESCRIPTION
**Pull request by zdover23**
This commit (finally) alphabetizes the terms in the glossary. This is not a grammar-correcting or usage-correcting commit.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -14,186 +14,193 @@ reflect either technical terms or legacy ways of referring to Ceph systems.
...
+                Versions of Ceph that have not yet been put through quality
+                assurance testing, but may contain new features.
 
-	Object Storage Device
-	OSD
-		A physical or logical storage unit (*e.g.*, LUN).
-		Sometimes, Ceph users use the
-		term "OSD" to refer to :term:`Ceph OSD Daemon`, though the
-		proper term is "Ceph OSD".
+	Ceph Test Framework
```
Comment (colemitchell) [inline doc/glossary.rst:54]: @zdover23 This appears to be in the wrong place.

File: doc/glossary.rst [hunk 2]
```diff
@@ -14,186 +14,193 @@ reflect either technical terms or legacy ways of referring to Ceph systems.
...
-		OSD and it is found in the OSD path in a file called ``osd_fsid``. This
-		``fsid`` term is used interchangeably with ``uuid``
+	Ceph Project
+                The aggregate term for the people, software, mission and
+                infrastructure of Ceph.
 
-	OSD uuid
-		Just like the OSD fsid, this is the OSD unique identifier and is used
-		interchangeably with ``fsid``
+	Ceph Platform
```
Comment (colemitchell) [inline doc/glossary.rst:86]: @zdover23 This appears to be in the wrong place.

File: doc/glossary.rst [hunk 3]
```diff
@@ -14,186 +14,193 @@ reflect either technical terms or legacy ways of referring to Ceph systems.
...
+                assurance testing, but may contain new features.
 
-	Object Storage Device
-	OSD
-		A physical or logical storage unit (*e.g.*, LUN).
-		Sometimes, Ceph users use the
-		term "OSD" to refer to :term:`Ceph OSD Daemon`, though the
-		proper term is "Ceph OSD".
+	Ceph Test Framework
+	Ceph Kernel Modules
```
Comment (colemitchell) [inline doc/glossary.rst:54]: @zdover23 I've found two alphabetization errors.

File: doc/glossary.rst (hunk 1 above)
Comment (zdover23) [reply doc/glossary.rst:54]: Thanks. I put it in the right place.

File: doc/glossary.rst (hunk 2 above)
Comment (zdover23) [reply doc/glossary.rst:86]: @colemitchell I've put it in the right place.

**Review by colemitchell — APPROVED** [review]

**Pull request merged**